File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/BookingService.java
package com.skedgo.tripkit.booking;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import io.reactivex.Flowable;

public interface BookingService {
  Flowable<BookingForm> getFormAsync(@NonNull String url);
  Flowable<BookingForm> postFormAsync(@NonNull String url, @Nullable InputForm inputForm);
  Flowable<BookingForm> postActionInputAsync(@NonNull String url, @NonNull String field, @NonNull String value);
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/BookingService.kt
package com.skedgo.tripkit.booking

import io.reactivex.Flowable

interface BookingService {
    fun getFormAsync(url: String): Flowable<BookingForm>

    fun postFormAsync(url: String, inputForm: InputForm?): Flowable<BookingForm>

    fun postActionInputAsync(url: String, field: String, value: String): Flowable<BookingForm>
}
